### PR TITLE
Set metadata in packrat lockfile

### PR DESCRIPTION
**Pull request by cderv**
Hey, 

This is related #425. @kevinushey I followed you advice and made one function to change different metadata. Just two for the moment `repos` and `r_version`. I think the others are not useful to modify. 

Can you take a look and tell me if the direction I took is ok ? 

I tried to write some tests but I don't manage to make them work. It seems like other test are not working either. 
I tried something like that
```r
projRoot <- cloneTestProject("healthy")
  init(enter = FALSE, projRoot, options = list(local.repos = "packages"))
``` 
but `init` does not work as expected.
help appreciated for those tests. thanks.

---
This change is [Reviewable](https://reviewable.io/reviews/rstudio/packrat/429)

### CODE REVIEW

File: R/lockfile-metadata.R [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#' # changes R version
+#' rver <- old_rver <- get_lockfile_metadata("r_version")
+#' rver
+#' rver <- "3.4.1"
+#' set_lockfile_metadata(r_version = rver)
+#' get_lockfile_metadata("r_version")
+#' # Setting back old state
+#' # set_lockfile_metadata(r_version = old_rver)
+#' }
+set_lockfile_metadata <- function(repos = NULL, r_version = NULL, project = NULL) {
```
Comment (kevinushey) [inline R/lockfile-metadata.R]: Most of the Packrat APIs take the project as the first argument -- can we follow that convention here?

File: R/lockfile-metadata.R [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  project <- getProjectDir(project)
+  lf_filepath <- lockFilePath(project)
+  if (!file.exists(lf_filepath)) {
+    stop(paste(lockFilePath, " is missing. Run packrat::init('",
+                 project, "') to generate it.", sep = ""))
+  }
+  lf <- as.data.frame(readDcf(lf_filepath), stringsAsFactors = F)
```
Comment (kevinushey) [inline R/lockfile-metadata.R]: Can we use `FALSE` instead of the abbreviated `F` here?

File: R/lockfile-metadata.R (hunk 1 above)
Comment (cderv) [reply R/lockfile-metadata.R]: I followed the convention of `set_opts` and `get_opts` that I find close in functionality. `project` is last in `get_opts` and second in `set_opts`. From usage :
```r
get_opts(options = NULL, simplify = TRUE, project = NULL)
set_opts(..., project = NULL, persist = TRUE)
```
bad idea ? 
Still want me to change ?

File: R/lockfile-metadata.R (hunk 2 above)
Comment (cderv) [reply R/lockfile-metadata.R]: Absolutely right ! Good practice still not automatic for me. Done!

File: R/lockfile-metadata.R (hunk 1 above)
Comment (kevinushey) [reply R/lockfile-metadata.R]: Good point! I think you're right; let's leave it as is then.

**Comment by kevinushey** [comment]
Thanks!

**Comment by cderv** [comment]
Oh thanks for merging! 
I thought it needed a test file but as I did not manage to make it work I did not push it. If you want me to still try for later do not hesitate to tell me! 
Should we add a bullet to NEWS.md too ? 
I also closed manually the related issue. Forgot to add closing keyword in the PR or commit message.

thanks again for you review !

**Comment by kevinushey** [comment]
I added a NEWS bullet and bumped the version here:

https://github.com/rstudio/packrat/commit/6ff1185e988331a58c04f2dd387e0a83e6355743

I'd welcome a PR adding tests as well if you have the time to put something together as well. Thanks for taking the time to put this together!

**Comment by cderv** [comment]
Thanks, should have looked in the other commit.. 
For the test, I tried to get inspiration from other tests but they are skipped on cran and does not seem to work anymore. I did not manage to created a dummy packrat project in temp folder.i will keep trying and I will come up with something eventually. You will welcome a PR when it is ready. No rush but will be better if tested.

**Pull request merged**